### DISCUSSION TITLE
ERR_STREAM_WRITE_AFTER_END on batched OData calls

### DISCUSSION THREAD

**Discussion by tsteckenborn** [mswjs/msw#1482] [Q&A] [answered]
Hey there,

Does anybody know a resolution to encountering the error 'ERR_STREAM_WRITE_AFTER_END'?

Calling the mocked APIs works fine as long as there's no batching involved. Once there's some sort of [batching](https://www.odata.org/documentation/odata-version-2-0/batch-processing/) involved it fails with the aforementioned error.

In case somebody of you already knows a workaround I'd be grateful for a solution.

**Comment by kettanaito** [comment, marked as answer]
Hey, @tsteckenborn. 

Could you create a reproduction repository that illustrates this issue? 

The error itself implies that something tries to write a response (I assume response here) body after the response stream has already ended. That's indeed a no-op. Whether that's an issue on the MSW's side, on the OData's side, or somewhere in-between is only discoverable with a reproduction repository.

**Reply by tsteckenborn** [reply]
We did some further debugging. Another middleware utilized with the framework seems to be causing the issue. Doesn't seem to be on your side.
